src/cwe_checker_lib/src/checkers/cwe_190.rs

```diff
@@ -105,13 +125,25 @@ pub fn check_cwe(
     cwe_params: &serde_json::Value,
 ) -> (Vec<LogMessage>, Vec<CweWarning>) {
     let project = analysis_results.project;
+    let pointer_inference_results = analysis_results.pointer_inference.unwrap();
+
     let config: Config = serde_json::from_value(cwe_params.clone()).unwrap();
     let mut cwe_warnings = Vec::new();
     let symbol_map = get_symbol_map(project, &config.symbols);
     for sub in project.program.term.subs.values() {
         for (block, jump, symbol) in get_callsites(sub, &symbol_map) {
             if block_contains_multiplication(block) {
-                cwe_warnings.push(generate_cwe_warning(&jump.tid, symbol));
+                let parms = match symbol.name.as_str() {
+                    "malloc" => vec![&symbol.parameters[0]],
+                    "xmalloc" => vec![&symbol.parameters[0]],
+                    "calloc" => vec![&symbol.parameters[0], &symbol.parameters[1]],
+                    "realloc" => vec![&symbol.parameters[1]],
+                    _ => vec![&symbol.parameters[0]],
```

Comment:
The general case should be to check all parameters for top values, since users might add other functions to the list where we do not know which parameters to check.. We can look up the list of all parameters through `project.program.term.extern_symbols`. And to nitpick a little more, we only need special cases for `realloc` (like you implemented it) and for `calloc` (where we should additionally check that the product of the two parameters does not overflow). But `malloc` and `xmalloc` do not need special cases. ;-)